controllers/tasks.js

```diff
@@ -347,6 +347,32 @@ const updateTaskStatus = async (req, res, next) => {
           return res.boom.badRequest("Status cannot be updated. Task is not done yet");
         }
       }
+      if (status) {
+        const isCurrentTaskStatusInProgress = task.taskData.status === TASK_STATUS.IN_PROGRESS;
+        const isCurrentTaskStatusBlock = task.taskData.status === TASK_STATUS.BLOCKED;
+        const isNewTaskStatusInProgress = status === TASK_STATUS.IN_PROGRESS;
+        const isNewTaskStatusBlock = status === TASK_STATUS.BLOCKED;
+        const isCurrProgress100 = parseInt(task.taskData.percentCompleted || 0) === 100;
+        const isCurrProgress0 = parseInt(task.taskData.percentCompleted || 0) === 0;
+        const isNewProgress100 = !!req.body.percentCompleted && parseInt(req.body.percentCompleted) === 100;
+        const isNewProgress0 = !!req.body.percentCompleted !== undefined && parseInt(req.body.percentCompleted) === 0;
+        if (isCurrentTaskStatusInProgress && !isNewTaskStatusBlock && !isCurrProgress100 && !isNewProgress100) {
+          return res.boom.badRequest(
+            "The status of task can not be changed from In progress until progress of task is not 100%."
+          );
+        }
+
+        if (isCurrentTaskStatusBlock && !isNewTaskStatusInProgress && !isCurrProgress100 && !isNewProgress100) {
+          return res.boom.badRequest(
+            "The status of task can not be changed from Blocked until progress of task is not 100%."
+          );
+        }
+        if (isNewTaskStatusInProgress && !isCurrentTaskStatusBlock && !isCurrProgress0 && !isNewProgress0) {
+          return res.boom.badRequest(
```

Comment:
same here

Reply:
Same as above this controller is not used for updating task to backlog status